### PR TITLE
Feat(eos_designs): Added the support of mgmt_interface_settings

### PR DESCRIPTION
**Pull request by MaheshGSLAB**
## Change Summary

Added the support of mgmt_interface_settings

## Related Issue(s)

Fixes #<ISSUE ID>

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes
Added the support of mgmt_interface_settings.

```
# OOB (Out-of-Band) MGMT interface settings.
mgmt_interface_settings: 
  description: <str; default="OOB_MANAGEMENT">
  vrf: <str; default="MGMT">
  vrf_routing: <bool; default=False>

  lldp:
    transmit: <bool>
    receive: <bool>
    ztp_vlan: <int>
```

## How to test
Run molecule

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.arista.com/stable/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/extensions/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by Vibhu-gslab — COMMENTED** [review]
Try to achieve 100% coverage, rest looks good

**Comment by MaheshGSLAB** [comment]
> Try to achieve 100% coverage, rest looks good

This missing coverage is for shared_utils/mgmt.py for method `default_mgmt_protocol_vrf` which is not used anywhere. Now am not getting why pylint did not raise any error.

Ok so pylint is not work like that and it do not catch the unused methods. There is tool called `vulture` used to  get the unused methods. Right now i just removed that method

File: python-avd/pyavd/_eos_designs/structured_config/base/management_interface.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2023-2026 Arista Networks, Inc.
```
Comment (MaheshGSLAB) [inline python-avd/pyavd/_eos_designs/structured_config/base/management_interface.py:1]: Created new file because the base init file reach its line number length.

File: python-avd/pyavd/_eos_designs/shared_utils/mgmt.py [hunk 2]
```diff
@@ -93,16 +112,6 @@ def default_mgmt_method(self: SharedUtilsProtocol) -> str | None:
 
         return None
 
-    @cached_property
-    def default_mgmt_protocol_vrf(self: SharedUtilsProtocol) -> str | None:
```
Comment (laxmikantchintakindi) [inline python-avd/pyavd/_eos_designs/shared_utils/mgmt.py:97]: Is this moved to some other place?

Comment (MaheshGSLAB) [reply python-avd/pyavd/_eos_designs/shared_utils/mgmt.py:97]: no this was never used

**Comment by laxmikantchintakindi** [comment]
There are some mentions of mgmt_interface_vrf in variable inputs of vrf. Need to fix that.

File: python-avd/pyavd/_eos_designs/schema/schema_fragments/mgmt_interface.schema.yml [hunk 3]
```diff
@@ -12,3 +12,8 @@ keys:
     type: str
     default: Management1
     description: OOB Management interface.
+    deprecation:
```
Comment (Vibhu-gslab) [inline python-avd/pyavd/_eos_designs/schema/schema_fragments/mgmt_interface.schema.yml:15]: We should add testcases for deprecated models in `eos_designs_deprecated_vars`

File: python-avd/pyavd/_eos_designs/shared_utils/mgmt.py [hunk 4]
```diff
@@ -41,6 +41,7 @@ def mgmt_interface(self: SharedUtilsProtocol) -> str:
             # Notice that we actually have a default value for the next two, but the precedence order would break if we use it.
             # TODO: Evaluate if we should remove the default values from either or both.
             self.platform_settings._get("management_interface", None),
+            self.inputs.mgmt_interface_settings._get("interface", None),
```
Comment (laxmikantchintakindi) [inline python-avd/pyavd/_eos_designs/shared_utils/mgmt.py:44]:
```suggestion
            self.inputs.mgmt_interface_settings._get_defined_attr("interface", None),
```

Comment (laxmikantchintakindi) [reply python-avd/pyavd/_eos_designs/shared_utils/mgmt.py:44]: Not sure this suggestion works, but using _get_defined_attr is more accurate here.

Comment (MaheshGSLAB) [reply python-avd/pyavd/_eos_designs/shared_utils/mgmt.py:44]: type hint is not happy with this

File: python-avd/pyavd/_eos_designs/schema/schema_fragments/mgmt_interface.schema.yml (hunk 3 above)
Comment (MaheshGSLAB) [reply python-avd/pyavd/_eos_designs/schema/schema_fragments/mgmt_interface.schema.yml:15]: added

**Review by laxmikantchintakindi — APPROVED** [review]

**Review by Vibhu-gslab — APPROVED** [review]

File: ansible_collections/arista/avd/extensions/molecule/eos_designs_unit_tests/inventory/host_vars/mgmt-interface-settings-1.yml
Comment (carl-baillargeon) [inline ansible_collections/arista/avd/extensions/molecule/eos_designs_unit_tests/inventory/host_vars/mgmt-interface-settings-1.yml:1]: Can you add a test with no lldp to test the empty-dict gate? Also in the same test, let's use `ipv6_mgmt_ip` on the node for completeness.

File: python-avd/pyavd/_eos_designs/schema/schema_fragments/mgmt_interface_settings.schema.yml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    description: |-
+      Out-of-Band (OOB) management interface settings.
+      These settings override the corresponding global variables where applicable.
+    type: dict
+    keys:
+      description:
+        type: str
+        description: |-
+          Interface description configured on the management interface.
+          Takes precedence over the global 'mgmt_interface_description' variable.
```
Comment (carl-baillargeon) [inline python-avd/pyavd/_eos_designs/schema/schema_fragments/mgmt_interface_settings.schema.yml]: I don't think we need the "Takes precedence over the global X variable". These keys are deprecated. The deprecation warning on the old keys already directs users to the new one.

File: ansible_collections/arista/avd/extensions/molecule/eos_designs_unit_tests/inventory/host_vars/mgmt-interface-settings-1.yml
Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/extensions/molecule/eos_designs_unit_tests/inventory/host_vars/mgmt-interface-settings-1.yml:1]: added

File: python-avd/pyavd/_eos_designs/schema/schema_fragments/mgmt_interface_settings.schema.yml (hunk 5 above)
Comment (MaheshGSLAB) [reply python-avd/pyavd/_eos_designs/schema/schema_fragments/mgmt_interface_settings.schema.yml]: updated

**Review by carl-baillargeon — APPROVED** [review]
